lib/rules/enforce-props-naming-consistency.js

```
@@ -0,0 +1,145 @@
+                        data: { paramName: param.name },
+                        fix: (fixer) => fixParameterName(fixer, param),
+                    });
+                }
+            }
+        }
+        // Check class constructor parameters
+        function checkClassConstructor(node) {
+            if (node.kind !== 'constructor') {
+                return;
+            }
+            const constructor = node.value;
```

Biome (2.1.2)

[error] Do not shadow the global "constructor" property.

Consider renaming this variable. It's easy to confuse the origin of variables when they're named after a known global.

(lint/suspicious/noShadowRestrictedNames)